### PR TITLE
fix compiling with c++17 on linux

### PR DESCRIPTION
**Pull request by arthurdead**

### CODE REVIEW

File: vphysics_jolt/cbase.h [hunk 1]
```diff
@@ -41,7 +41,7 @@
 #include "tier1/tier1.h"
 #include "tier1/strtools.h"
 #include "tier1/interface.h"
-#include "tier1/keyvalues.h"
+#include "tier1/KeyValues.h"
```
Comment (Joshua-Ashton) [inline vphysics_jolt/cbase.h:44]: Hm, in our code base for Desolation this is lowercase. Ugh.

File: vphysics_jolt/vjolt_objectpairhash.cpp [hunk 2]
```diff
@@ -71,7 +76,11 @@ void JoltPhysicsObjectPairHash::RemoveAllPairsForObject( void *pObject0 )
 
 bool JoltPhysicsObjectPairHash::IsObjectInHash( void *pObject0 )
 {
+#if __cplusplus >= 202002L
     return m_Objects.contains( pObject0 );
+#else
+	return m_Objects.find( pObject0 ) != m_Objects.cend();
```
Comment (Joshua-Ashton) [inline vphysics_jolt/vjolt_objectpairhash.cpp:82]: Shouldn't this be just .end()?

File: vphysics_jolt/vjolt_objectpairhash.cpp [hunk 3]
```diff
@@ -54,7 +54,12 @@ void JoltPhysicsObjectPairHash::RemoveObjectPair( void *pObject0, void *pObject1
 bool JoltPhysicsObjectPairHash::IsObjectPairInHash( void *pObject0, void *pObject1 )
 {
     auto pair = CreateSortedPair( pObject0, pObject1 );
-    return m_PairHashes[ GetHashArrayIndex( PointerHasher{}( pair ) ) ].contains( pair );
+	auto &pairHashes = m_PairHashes[ GetHashArrayIndex( PointerHasher{}( pair ) ) ];
+#if __cplusplus >= 202002L
+    return pairHashes.contains( pair );
+#else
+	return pairHashes.find( pair ) != pairHashes.cend();
```
Comment (Joshua-Ashton) [inline vphysics_jolt/vjolt_objectpairhash.cpp:61]: Same here

File: vphysics_jolt/vjolt_listener_contact.h [hunk 4]
```diff
@@ -5,6 +5,13 @@
 
 struct JoltPhysicsContactPair
 {
+#if __cplusplus < 202002L
```
Comment (Joshua-Ashton) [inline vphysics_jolt/vjolt_listener_contact.h:8]: Probably dont need to ifdef it

File: vphysics_jolt/vjolt_controller_fluid.cpp [hunk 5]
```diff
@@ -103,7 +103,14 @@ void JoltPhysicsFluidController::OnJoltPhysicsObjectDestroyed( JoltPhysicsObject
 	if ( pObject == m_pFluidObject )
 		m_pFluidObject = nullptr;
 
+#if __cplusplus >= 202002L
 	std::erase_if( m_ObjectsInShape, [pObject]( JoltPhysicsObject *pCachedObject ) { return pObject == pCachedObject; } );
+#else
+	for ( auto it( m_ObjectsInShape.begin() ); it != m_ObjectsInShape.end(); ++it ) {
+		if ( *it == pObject )
+			it = m_ObjectsInShape.erase(it);
```
Comment (Joshua-Ashton) [inline vphysics_jolt/vjolt_controller_fluid.cpp:111]: Can break here, no need to iterate further

**Comment by Joshua-Ashton** [comment]
Please also fix indentation, we use 1 tab

**Comment by arthurdead** [comment]
> Please also fix indentation, we use 1 tab

i did use 1 tab its vjolt_objectpairhash.cpp that uses 4 spaces, i changed it to 4 spaces to match

File: vphysics_jolt/vjolt_objectpairhash.cpp (hunk 2 above)
Comment (arthurdead) [reply vphysics_jolt/vjolt_objectpairhash.cpp]: i mean it doenst matter, the funcs should have been marked as const but they arent so i used cend

**Comment by Joshua-Ashton** [comment]
Looks like GitHub PR reviewing always shows with broken indentation too on other files for me :(

File: vphysics_jolt/vjolt_objectpairhash.cpp [hunk 6]
```diff
@@ -54,7 +54,12 @@ void JoltPhysicsObjectPairHash::RemoveObjectPair( void *pObject0, void *pObject1
 bool JoltPhysicsObjectPairHash::IsObjectPairInHash( void *pObject0, void *pObject1 )
 {
     auto pair = CreateSortedPair( pObject0, pObject1 );
-    return m_PairHashes[ GetHashArrayIndex( PointerHasher{}( pair ) ) ].contains( pair );
+	auto &pairHashes = m_PairHashes[ GetHashArrayIndex( PointerHasher{}( pair ) ) ];
+#if __cplusplus >= 202002L
```
Comment (Joshua-Ashton) [inline vphysics_jolt/vjolt_objectpairhash.cpp:58]: Sorry I wasn't clear, I think we should drop all these CPP ifdefs

File: vphysics_jolt/cbase.h (hunk 1 above)
Comment (Joshua-Ashton) [reply vphysics_jolt/cbase.h:44]: Can we put this behind the `#ifndef GAME_L4D2_OR_NEWER` that I just added to main?
Looks like that's the first branch where this casing changed.

File: vphysics_jolt/vjolt_listener_contact.h [hunk 7]
```diff
@@ -5,6 +5,11 @@
 
 struct JoltPhysicsContactPair
 {
+	JoltPhysicsContactPair( JoltPhysicsObject *object1, JoltPhysicsObject *object2 )
```
Comment (Joshua-Ashton) [inline vphysics_jolt/vjolt_listener_contact.h:8]: nit: pObject1 pObject2

File: vphysics_jolt/vjolt_controller_fluid.cpp [hunk 8]
```diff
@@ -103,7 +103,16 @@ void JoltPhysicsFluidController::OnJoltPhysicsObjectDestroyed( JoltPhysicsObject
 	if ( pObject == m_pFluidObject )
 		m_pFluidObject = nullptr;
 
+#if __cplusplus >= 202002L
 	std::erase_if( m_ObjectsInShape, [pObject]( JoltPhysicsObject *pCachedObject ) { return pObject == pCachedObject; } );
+#else
+	for ( auto it( m_ObjectsInShape.begin() ); it != m_ObjectsInShape.end(); ++it ) {
```
Comment (Joshua-Ashton) [inline vphysics_jolt/vjolt_controller_fluid.cpp:106]:
```cpp
for ( auto it = m_ObjectsInShape.begin(); it != m_ObjectsInShape.end(); it++ )
{
	if ( *it == pObject )
	{
		m_ObjectsInShape.erase( it );
		break;
	}
}
```

Comment (Joshua-Ashton) [reply vphysics_jolt/vjolt_controller_fluid.cpp:106]: (remove ifdef too)

File: vphysics_jolt/vjolt_objectpairhash.cpp [hunk 9]
```diff
@@ -71,7 +76,11 @@ void JoltPhysicsObjectPairHash::RemoveAllPairsForObject( void *pObject0 )
 
 bool JoltPhysicsObjectPairHash::IsObjectInHash( void *pObject0 )
 {
+#if __cplusplus >= 202002L
     return m_Objects.contains( pObject0 );
+#else
```
Comment (Joshua-Ashton) [inline vphysics_jolt/vjolt_objectpairhash.cpp]: remove ifdef

**Comment by Joshua-Ashton** [comment]
Okay, more nits then LGTM! :D

Sorry for all the style nit picks, I just want to keep the code as clean and readable as possible.

File: vphysics_jolt/cbase.h (hunk 1 above)
Comment (Joshua-Ashton) [reply vphysics_jolt/cbase.h:44]: Actually I will just do this myself to spare you the pain of rebasing.

**Pull request merged**